### PR TITLE
Category Slidedown Testing

### PR DESCRIPTION
**Pull request by rgomezp**
# Description
Testing for Category Slidedown. Includes some refactoring and other tweaks.

# Unit Testing
### PromptsManager.ts
- 'if category options are not configured, check that an error was logged'
- 'category options are configured, in update mode, check 1) tagging container loaded 2) remote tag fetch'
- 'category options are configured, not in update mode, check remote tag fetch not made'

### TagManager.ts
- 'calling `syncTags` results in remote tag update with sendTags'
- '`downloadTags` results in remote getTags call'
- 'subscribing through category slidedown accept button calls syncTags'

### TaggingContainer.ts
- 'check sanitization is working correctly'
- 'check that correct TagCategory object is returned after applying remote player tags'

### TagUtils.ts
- 'check conversion from tag number values to booleans'
- 'check conversion from tag boolean values to numbers'
- 'check tagHelperWithRetries makes correct number of tries'

# Refactoring
- Renamed unit tests to prevent duplicates
- `setupOneSignalWithStubs()` helper replaces `beforeTest()`

This change is [Reviewable](https://reviewable.io/reviews/onesignal/onesignal-website-sdk/659)

### CODE REVIEW

File: src/slidedown/TaggingContainer.ts [hunk 1]
```diff
@@ -34,14 +40,25 @@ export default class TaggingContainer {
...
+    /**
+     * Returns checked TagCategory[] using unchecked remoteTagCategories (from config)
+     * and existingPlayerTags (from `getTags`)
+     * @param  {TagCategory[]} remoteTagCategories
+     * @param  {TagsObject} existingPlayerTags?
+     */
+    private _getCheckedTagCategories(remoteTagCategories: TagCategory[], existingPlayerTags?: TagsObject)
+        : TagCategory[] {
+            const remoteTagCategoriesCopy: TagCategory[] = JSON.parse(JSON.stringify(remoteTagCategories));
+            return !!existingPlayerTags ?
```
Comment (jkasten2) [inline src/slidedown/TaggingContainer.ts:52]: Make this an if instead

File: src/slidedown/TaggingContainer.ts [hunk 2]
```diff
@@ -81,6 +98,8 @@ export default class TaggingContainer {
         }
     }
 
+    // static
```
Comment (jkasten2) [inline src/slidedown/TaggingContainer.ts:101]: Remove comment

File: src/slidedown/TaggingContainer.ts [hunk 3]
```diff
@@ -61,12 +78,12 @@ export default class TaggingContainer {
 
     private getCategoryLabelHtml(tagCategory: TagCategory): string {
         const { label } = tagCategory;
-        return `<label class="onesignal-category-label" title="${sanitizeHtml(label)}">
-        <span class="onesignal-category-label-text">${sanitizeHtml(label)}</span>
-        <input type="checkbox" value="${sanitizeHtml(tagCategory.tag)}"
-            ${tagCategory.checked ? `checked="${sanitizeHtml(tagCategory.checked)}` : ``}">
-        <span class="onesignal-checkmark"></span></label>
-        <div style="clear:both"></div>`;
+        return `<label class="onesignal-category-label" title="${sanitizeHtmlAndDoubleQuotes(label)}">`+
```
Comment (jkasten2) [inline src/slidedown/TaggingContainer.ts:81]: Space before +

File: test/unit/utils/TagUtils.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    const converted = TagUtils.convertTagNumberValuesToBooleans({ tag1: "1", tag2: "0" });
+    t.true(_.isEqual(converted, { tag1: true, tag2: false }));
+});
+
+test('check conversion from tag boolean values to numbers', t => {
+    const converted = TagUtils.convertTagBooleanValuesToNumbers({ tag1: false, tag2: true });
+    t.true(_.isEqual(converted, { tag1: 0, tag2: 1 }));
+});
+
+test('check tagHelperWithRetries makes correct number of tries', async t => {
```
Comment (jkasten2) [inline test/unit/utils/TagUtils.ts:30]: We should remove `tagHelperWithRetries` from the source as well if this is going away. We can make a comment pointing to the comment if we want to add back the code later.

**Comment by rgomezp** [comment]
Cleaned up PR: https://github.com/OneSignal/OneSignal-Website-SDK/pull/665

**Pull request closed without merging**